### ISSUE TITLE
Questions for inference phase

### ISSUE THREAD

**Issue by JiahaoChen1** [ZhaoningYu1996/HM-GNN#7]
Hi, thanks for your great work! But i am a little confusing about inference phase. Since the Heterogeneous motif graph is built by the training set, how to make inference with unseen molecules in test set ? How to make these molecules as nodes in Heterogeneous motif graph?

**Comment by ZhaoningYu1996** [comment]
Hi Jiahao,

Thank you for being so interested in our work.
In our paper, we are using semi-supervised learning to evaluate our method, which means both the training and test sets are in the heterogeneous motif graph. In the training step, we only use the training set to train, while in the testing step, we use the testing set.

You can also split training and testing sets by keeping the motif nodes in the heterogeneous graph and adding different molecule nodes in different steps.

Hope this help.
Feel free to let me know if you have any other questions.

**Comment by JiahaoChen1** [comment]
Thanks! More questions are how to guarantee the motif vocabulary to cover the test molecule. Does any molecule can be represented by a set of motifs from the given motif vocabulary?

**Comment by ZhaoningYu1996** [comment]
In our paper, motifs are generated from the whole dataset, which can guarantee the motif vocabulary to cover the test molecule.
You can also choose to generate motifs from the training set. The results should be similar.

**Comment by JiahaoChen1** [comment]
OK, thanks for your response!
